scripts/content-utils/card-tree.ts

```diff
@@ -66,10 +75,18 @@ export function cardsFromTree(tree: CardTree, bindRef?: string): CardData[] {
                             ...triggerRefRemoval,
                         },
                     ),
-                ),
+                ]),
             ],
         ),
-        ...(tree.left ? cardsFromTree(tree.left, leftRef) : []),
-        ...(tree.right ? cardsFromTree(tree.right, rightRef) : []),
+        ...(tree.left && "card" in tree.left
+            ? cardsFromTree(tree.left, leftRef)
+            : []),
+        ...(tree.right && "card" in tree.right
+            ? cardsFromTree(tree.right, rightRef)
+            : []),
     ]
 }
+
+function mixToArray<T>(data?: T | T[]): T[] {
+    return data ? (Array.isArray(data) ? data : [data]) : []
```

Comment:
Some generic wizardry right here. But after some thought this is really handy. Should probably be exported as a util function right? I know some other places where we're doing similar things to use the `(data: T | T[]) => T[]` transformation.